### ISSUE TITLE
Not working on firefox

### ISSUE THREAD

**Issue by seanparko** [flackr/scroll-timeline#254]
I've tried even running the demos provided in firefox, but none of them seem to work. The instructions dont seem to really explain how to implement it either. I just want to css the css scroll() function. Do I just have to add the script tag to my html head?

**Comment by seanparko** [comment]
*It works with inline styles ONLY. This video explains how to get css animations working correctly with this polyfill in firefox: https://www.youtube.com/watch?v=UmzFk68Bwdk. Still not sure why the demos don't work.

**Comment by tatof** [comment]
Oke, is it possible to use Polyfill only on one specific .css file?
Its impossible to remove @font-face or any other Media Query

**Comment by bramus** [comment]
See #248 for that.

**Comment by seanparko** [comment]
> > *It works with inline styles ONLY
> 
> This used to be the case but no longer is since early February, when #242 to merged. See https://flackr.github.io/scroll-timeline/demo/view-timeline-external-css/ for a demo.

Just tested that exact demo in firefox and chrome. It didn't seem to work in firefox, only chrome.

**Comment by bramus** [comment]
Are you using Firefox Nightly by any chance? They have a half-working implementation enabled in Nightly, so the polyfill doesn’t kick in there as Firefox claims support.

Be sure to test in Firefox Stable.

**Comment by seanparko** [comment]
Just to confirm at this point all I have to do is just add the script to the head of my document, nothing else?

**Comment by Antonytm** [comment]
Doesn't work for me as well.
Next.js: Script was added to the <head> tag.

**Comment by bramus** [comment]
> Next.js: Script was added to the tag.

That is a problem caused by using Next.js. See #240 for that.

**Comment by seanparko** [comment]
> Are you using Firefox Nightly by any chance? They have a half-working implementation enabled in Nightly, so the polyfill doesn’t kick in there as Firefox claims support. Be sure to test in Firefox Stable.
> 
> When testing locally, be sure to build the project first.

Okay, I made sure I'm using base firefox. I have just added the following to the top of the head of my html, still can't get it to work. Do I need to change my CSS or something? Is there something else I need to do?
`<script src="https://flackr.github.io/scroll-timeline/dist/scroll-timeline.js"></script>`

**Comment by bramus** [comment]
At this point, sharing tour CSS and/or a repro is necessary to debug this.

**Comment by flackr** [comment]
What do you mean by "locally"? Are you loading directly from a file rather than a webserver? For external CSS files the polyfill needs to be able to `fetch` the CSS in order for the polyfill to read them. The browser does not allow this from file:/// origins as general access would allow reading other files from your system.

You will have to run your demo through a webserver, e.g.
```
python3 -m http.server
```

I confirmed that it looks like it's working in firefox for me using the above to serve the website and load it from http://localhost:8000/navbar.html

**Comment by seanparko** [comment]
> What do you mean by "locally"? Are you loading directly from a file rather than a webserver? For external CSS files the polyfill needs to be able to `fetch` the CSS in order for the polyfill to read them. The browser does not allow this from file:/// origins as general access would allow reading other files from your system.
> 
> You will have to run your demo through a webserver, e.g.
> 
> ```
> python3 -m http.server
> ```
> 
> I confirmed that it looks like it's working in firefox for me using the above to serve the website and load it from http://localhost:8000/navbar.html

Okay, I tried running it on my WAMP server, and the code appears to work now. However, I'm still having trouble with it getting to work with certain features. If you go back to my github repo, I have updated the code so that it contains something that I cannot get to work. When you scroll down, the SVG in the html file is suppose to grow with respect to the scroll position. Again, it works perfectly on chrome but not on firefox. Thanks for your help so far.

https://github.com/seanparko/boilerplate_site

**Comment by seanparko** [comment]
I just said screw it and decided to push my site to the public. I've now noticed that the console log is absolutely filled with errors regarding the script on firefox. Here is an example: "
ncaught TypeError: Animation.currentTime setter: Value being assigned is not a finite floating-point value.
    ct proxy-animation.js:481
    lt proxy-animation.js:461
    st proxy-animation.js:403
    et proxy-animation.js:180
    mt proxy-animation.js:657
    mt proxy-animation.js:655
    xe scroll-timeline-base.js:44
    o scroll-timeline-base.js:311
proxy-animation.js:481
"

**Comment by seanparko** [comment]
@flackr @bramus Bump. Have any of you been able to take a look at that site I uploaded and recreate the issue? Could it be to do with relative units?

**Comment by jpike88** [comment]
>Doesnt work in firefox with inline styles AND breaks @font-face for some reason

This is an insanely annoying bug to have to track down, I've wasted a lot of time for just pretty scrollbars.

I also was getting the `Value being assigned is not a finite floating-point value` error.

I also don't appreciate a 3rd party lib deciding to fetch a js file from a GitHub hosted resource... even when bundled for production.

I've decided to fork ngx-scrollbar and ditch the polyfill code.

**Comment by jpike88** [comment]
Just found out about the `NG_SCROLLBAR_POLYFILL` token, I used that instead and hosted the polyfill file locally.

**Comment by jpike88** [comment]
Nope, that still breaks Firefox's font-face loading functionality for some reason.

**Comment by jpike88** [comment]
So after reading up, the CSS parser is mangling stuff and causing Firefox to break any font-face implementations? That is a pretty huge caveat, I guess it's back to native scrollbars and thinking about an alternative library.

**Comment by seanparko** [comment]
For me the polyfill its pretty much useless. Has too many limitations (Must use non-relative units, can't have font-face, etc). Would be nice if the readme of the git was updated so many afternoons weren't wasted trying to get it to work. Guess we'll have to wait till its implemented into all browsers.

**Comment by MurhafSousli** [comment]
> Just found out about the `NG_SCROLLBAR_POLYFILL` token, I used that instead and hosted the polyfill file locally.

@jpike88  In v14.1.1 the polyfill is shipped with the ngx-scrollbar, but it still load it directly from github by default, to load it locally, read here  https://github.com/MurhafSousli/ngx-scrollbar/wiki/Scroll-timeline-polyfill
